clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java

```diff
@@ -0,0 +1,219 @@
+  public ZeebeClientBuilder defaultJobTimeout(final Duration timeout) {
+    return innerBuilder.defaultJobTimeout(timeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobPollInterval(final Duration pollInterval) {
+    return innerBuilder.defaultJobPollInterval(pollInterval);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultMessageTimeToLive(final Duration timeToLive) {
+    return innerBuilder.defaultMessageTimeToLive(timeToLive);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultRequestTimeout(final Duration requestTimeout) {
+    return innerBuilder.defaultRequestTimeout(requestTimeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder caCertificatePath(final String certificatePath) {
+    return innerBuilder.caCertificatePath(certificatePath);
+  }
+
+  @Override
+  public ZeebeClientBuilder keepAlive(final Duration keepAlive) {
+    return innerBuilder.keepAlive(keepAlive);
+  }
+
+  @Override
+  public ZeebeClientBuilder withInterceptors(final ClientInterceptor... interceptor) {
+    return innerBuilder.withInterceptors(interceptor);
+  }
+
+  @Override
+  public ZeebeClientBuilder withJsonMapper(final JsonMapper jsonMapper) {
+    return innerBuilder.withJsonMapper(jsonMapper);
+  }
+
+  @Override
+  public ZeebeClient build() {
+    innerBuilder.gatewayAddress(determineGatewayAddress());
+    innerBuilder.credentialsProvider(determineCredentialsProvider());
+    return innerBuilder.build();
+  }
+
+  private String determineGatewayAddress() {
+    if (isNeedToSetCloudGatewayAddress()) {
+      ensureNotNull("cluster id", clusterId);
+      return String.format("%s.%s:443", clusterId, BASE_ADDRESS);
+    } else {
+      Loggers.LOGGER.debug(
+          "Expected that the gateway address is not set, "
+              + "but got this gateway address: {} set in the client cloud builder; "
+              + "should be not set if you are trying to connect to the Camunda Cloud.",
+          innerBuilder.getGatewayAddress());
+      return innerBuilder.getGatewayAddress();
+    }
+  }
+
+  private CredentialsProvider determineCredentialsProvider() {
+    final CredentialsProvider provider = innerBuilder.getCredentialsProvider();
+    if (provider == null) {
+      ensureNotNull("cluster id", clusterId);
+      ensureNotNull("client id", clientId);
+      ensureNotNull("client secret", clientSecret);
+      final OAuthCredentialsProviderBuilder builder = new OAuthCredentialsProviderBuilder();
+      return builder
+          .audience(String.format("%s.%s", clusterId, BASE_ADDRESS))
+          .clientId(clientId)
+          .clientSecret(clientSecret)
+          .authorizationServerUrl(BASE_AUTH_URL)
+          .build();
+    } else {
+      Loggers.LOGGER.debug(
+          "Expected that the credentials provider is not set, "
+              + "but got this credentials provider: {} set in the client cloud builder; "
+              + "should be not set if you are trying to authenticate against the Camunda Cloud.",
```

Comment:
A user can create their own credentials provider that works correctly with Camunda Cloud. In addition, the provider does not implement toString nicely. I recommend changing this to:
`Expected to use 'cluster id', `client id` and `client secret` to set credentials provider in the client cloud builder, but overwriting with explicitly defined credentials provider`